test/integration/kubernetes_deploy_test.rb

```diff
@@ -1053,4 +1053,44 @@ def test_raise_on_yaml_missing_kind
       "  datapoint: value1"
     ], in_order: true)
   end
+
+  def test_hpa_can_be_successful
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    deploy_metric_server
+    assert_deploy_success(deploy_fixtures("hpa/hpa"))
+    assert_logs_match_all([
+      "Deploying resources:",
+      "HorizontalPodAutoscaler/hello-hpa (timeout: 300s)",
+      %r{HorizontalPodAutoscaler/hello-hpa\s+Succeeded}
+    ])
+  end
+
+  def test_hpa_can_fail_when_no_matching_deployment
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    deploy_metric_server
+    assert_deploy_failure(deploy_fixtures("hpa/hpa", subset: ["hpa.yml"]))
+    assert_logs_match_all([
+      "Deploying HorizontalPodAutoscaler/hello-hpa (timeout: 300s)",
+      "HorizontalPodAutoscaler/hello-hpa: FAILED",
+      "Final status: FailedGetScale"
+    ])
+  end
+
+  def test_hpa_can_be_pruned
+    skip if KUBE_SERVER_VERSION < Gem::Version.new('1.8.0')
+    deploy_metric_server
+    assert_deploy_success(deploy_fixtures("hpa/hpa"))
+    assert_deploy_success(deploy_fixtures("hpa/hpa", subset: ["deployment.yml"]))
+    assert_logs_match_all(['The following resources were pruned: horizontalpodautoscaler "hello-hpa"'])
+  end
+
+  private
+
+  def deploy_metric_server
+    # Set-up the metric server that the HPA needs https://github.com/kubernetes-incubator/metrics-server
+    ns = @namespace
+    @namespace = "kube-system"
+    assert_deploy_success(deploy_fixtures("hpa/kube-system", allow_protected_ns: true, prune: false))
```

Comment:
`hpa/kube-system` isn't really a fixture set; it's more like test infra components. I'd move all those configs somewhere else, like `test/setup/metrics-server` or something. Maybe we should even use KubeClient or `kubectl apply` to create it... seems a little weird to have the test setup itself depend on the correct functioning of the core code.